src/v/storage/types.cc

```diff
@@ -74,7 +74,15 @@ std::ostream& operator<<(std::ostream& o, const timequery_result& a) {
     return o << "{offset:" << a.offset << ", time:" << a.time << "}";
 }
 std::ostream& operator<<(std::ostream& o, const timequery_config& a) {
-    return o << "{max_offset:" << a.max_offset << ", time:" << a.time << "}";
+    o << "{max_offset:" << a.max_offset << ", time:" << a.time
+      << ", type_filter:";
+    if (a.type_filter) {
```

Comment:
nit: we have optional print operator in `src/v/utils/to_string.h`

Reply:
Good to know, I copied this from `log_reader_config`.